plugins/pencilblue/controllers/actions/admin/content/media/delete_media.js

```diff
@@ -77,7 +77,7 @@ module.exports = function(pb) {
     };
 
     DeleteMediaController.prototype.removeLocal = function(media, mservice, cb) {
-        if (!media.is_file) {
+        if (!media.is_file && media.location.indexOf('http') !== 0 && media.location.indexOf('//') !== 0) {
```

Comment:
The provider's implemented delete method would receive the absolute url contained in the media.location.  The provider then issues an HTTP DELETE against the absolute url provided to remove it.  Alternatively, might there be a way to facilitate the provider to toggle the is_file property boolean to their liking?  This would then eliminate the need to check for an absolute url.